cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -233,6 +235,41 @@ struct DivideChecked {
   }
 };
 
+struct Exponentiate {
+  template <typename T, typename Arg0, typename Arg1>
+  static enable_if_integer<T> Call(KernelContext* ctx, Arg0 left, Arg1 right) {
+    if (left == 0 && (-INFINITY < right && right < 0)) {
+      ctx->SetStatus(Status::Invalid("divide by zero"));
```

Comment:
For reference, error message from numpy for integer power:
```
In [2]: np.power(1, -1)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-eff91c77f984> in <module>
----> 1 np.power(1, -1)

ValueError: Integers to negative integer powers are not allowed.

```

Reply:
I'll make these match numpy.

Reply:
See the other comment.